### PR TITLE
Update azure_rm_networkinterface related document

### PR DESCRIPTION
**Pull request by Fred-sun**
##### SUMMARY

Update azure_rm_networkinterface related document.

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

azure_rm_networkinterface/azure_rm_networkinterface_facts
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 1]
```diff
@@ -48,32 +49,30 @@
...
         required: true
     os_type:
         description:
-            - Determines any rules to be added to a default security group. When creating a network interface, if no
-              security group name is provided, a default security group will be created. If the os_type is 'Windows',
-              a rule will be added allowing RDP access. If the os_type is 'Linux', a rule allowing SSH access will be
-              added.
+            - Determines any rules to be added to a default security group.
+            - When creating a network interface, if no security group name is provided, a default security group will be created.
+            - If the I(os_type=Windows), a rule will be added allowing RDP access.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]:
```suggestion
            - If the I(os_type=Windows), a rule allowing RDP access will be added.
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 2]
```diff
@@ -84,18 +83,18 @@
...
         default: Dynamic
         choices:
             - Dynamic
             - Static
     public_ip:
         description:
-            - (Deprecate) When creating a network interface, if no public IP address name is provided a default public IP
-              address will be created. Set to false, if you do not want a public IP address automatically created.
+            - (Deprecate) When creating a network interface, if no public IP address name is provided a default public IP address will be created.
+            - Set to C(false), if you do not want a public IP address automatically created.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]:
```suggestion
            - Set to C(false) if you do not want a public IP address automatically created.
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 3]
```diff
@@ -129,7 +128,7 @@
                     - Private ip address for the ip configuration.
             private_ip_allocation_method:
                 description:
-                    - private ip allocation method.
+                    - Private ip allocation method.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]:
```suggestion
                    - Private IP allocation method.
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 4]
```diff
@@ -142,7 +141,7 @@
                     - public_ip_name
             public_ip_allocation_method:
                 description:
-                    - public ip allocation method.
+                    - Public ip allocation method.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]:
```suggestion
                    - Public IP allocation method.
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 5]
```diff
@@ -84,18 +83,18 @@
             - This option will be deprecated in 2.9, use I(ip_configurations) instead.
     private_ip_allocation_method:
         description:
-            - "(Deprecate) Specify whether or not the assigned IP address is permanent. NOTE: when creating a network interface
-              specifying a value of 'Static' requires that a private_ip_address value be provided. You can update
-              the allocation method to 'Static' after a dynamic private ip address has been assigned."
+            - (Deprecate) Specify whether or not the assigned IP address is permanent.
+            - When creating a network interface specifying a value of C(Static) requires that a I(private_ip_address) value be provided.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]:
```suggestion
            - When creating a network interface, if you specify I(private_ip_allocation_method=Static), 
              you must provide a value for I(private_ip_address).
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 6]
```diff
@@ -108,8 +107,8 @@
             - public_ip_name
     public_ip_allocation_method:
         description:
-            - (Deprecate) If a public_ip_address_name is not provided, a default public IP address will be created. The allocation
-              method determines whether or not the public IP address assigned to the network interface is permanent.
+            - (Deprecate) If a public_ip_address_name is not provided, a default public IP address will be created.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]:
```suggestion
            - (Deprecate) If a I(public_ip_address_name) is not provided, a default public IP address will be created.
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 7]
```diff
@@ -118,7 +117,7 @@
     ip_configurations:
         description:
             - List of ip configuration if contains mutilple configuration, should contain configuration object include
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]:
```suggestion
            - List of ip configuration if contains multiple configuration, should contain configuration object include
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py [hunk 8]
```diff
@@ -161,59 +160,59 @@
...
         type: bool
         default: False
-        version_added: 2.7
+        version_added: "2.7"
     dns_servers:
         description:
-            - Which DNS servers should the NIC lookup
-            - List of IP's
+            - Which DNS servers should the NIC lookup.
+            - List of IP's.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py]:
```suggestion
            - List of IP addresses.
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py [hunk 9]
```diff
@@ -143,13 +145,13 @@
                         - Private ip address for the ip configuration.
                 private_ip_allocation_method:
                     description:
-                        - private ip allocation method.
+                        - Private ip allocation method.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py]:
```suggestion
                        - Private IP allocation method.
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py [hunk 10]
```diff
@@ -143,13 +145,13 @@
...
                 public_ip_address:
                     description:
                         - Name of the public ip address. None for disable ip address.
                 public_ip_allocation_method:
                     description:
-                        - public ip allocation method.
+                        - Public ip allocation method.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py]:
```suggestion
                        - Public IP allocation method.
```

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py [hunk 11]
```diff
@@ -175,8 +177,8 @@
                 - Whether to enable IP forwarding
         dns_servers:
             description:
-                - Which DNS servers should the NIC lookup
-                - List of IP's
+                - Which DNS servers should the NIC lookup.
+                - List of IP's.
```
Comment (acozine) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py]:
```suggestion
                - List of IP addresses.
```

**Review by acozine — COMMENTED** [review]
Thanks @Fred-sun for updating this documentation. I found a few typos and other small changes to suggest on this PR. 

I'm curious, why are you adding quotation marks around the `version_added` numbers? They display fine without them, and we've been removing them from other module docs. See this screenshot from the `devel` docs for `azure_rm_networkinterface`: 

Screen Shot 2019-06-10 at 10 23 17 AM

I'd recommend leaving the `version_added` numbers without `"` or `'` around them.

**Comment by Fred-sun** [comment]
> Thanks @Fred-sun for updating this documentation. I found a few typos and other small changes to suggest on this PR.
> I'm curious, why are you adding quotation marks around the version_added numbers? They display fine without them, and we've been removing them from other module docs. See this screenshot from the devel docs for azure_rm_networkinterface:
> 
> I'd recommend leaving the version_added numbers without " or ' around them.

Thanks for your review.  This is required from the format of the module, you can find the answer in this [link](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html)!
1.  Only needed if this option was extended after initial Ansible release, i.e. this is greater than the top level version_added field.
2.  This is a string, and not a float, i.e. version_added: '2.3'.

**Comment by Fred-sun** [comment]
ready_for_review

**Pull request merged**